be/src/olap/rowset/segment_v2/binary_plain_page.h

```diff
@@ -80,24 +80,21 @@ class BinaryPlainPageBuilder : public PageBuilder {
     Slice finish() override {
         _finished = true;
 
-        size_t offsets_pos = _buffer.size();
-
-        // Set up the header
-        encode_fixed32_le(&_buffer[0], _offsets.size());
-        encode_fixed32_le(&_buffer[4], offsets_pos);
+        // Set up trailer
+        size_t new_cap = _buffer.size() + _offsets.size() * sizeof(uint32_t) + MAX_TRAILER_SIZE;
+        if (new_cap > _buffer.capacity())
+            _buffer.GrowArray(new_cap);
```

Comment:
when append trailer to buffer_,the size is known,so grow buffer manually can avoid auto grow which may more than needed(auto grow always expand by at least 50%),this will waste memory

Reply:
we already reserve `_options.data_page_size` bytes ahead in reset(), so grow rarely happens

Reply:
👌